source/cell/cell.cpp

```diff
@@ -767,9 +814,115 @@ bool cell::has_format() const
+        cloned_format.alignment(src_alignment, source_format.d_->alignment_applied);
+    }
+
+    // Copy border if present
+    if (source_format.d_->border_id.is_set())
+    {
+        auto src_border = source_format.d_->parent->borders.at(source_format.d_->border_id.get());
+        cloned_format.border(src_border, source_format.d_->border_applied);
+    }
+
+    // Copy fill if present
+    if (source_format.d_->fill_id.is_set())
+    {
+        auto src_fill = source_format.d_->parent->fills.at(source_format.d_->fill_id.get());
+        cloned_format.fill(src_fill, source_format.d_->fill_applied);
+    }
+
+    // Copy font if present
+    if (source_format.d_->font_id.is_set())
+    {
+        auto src_font = source_format.d_->parent->fonts.at(source_format.d_->font_id.get());
+        cloned_format.font(src_font, source_format.d_->font_applied);
+    }
+
+    // Copy number format if present
+    if (source_format.d_->number_format_id.is_set())
+    {
+        const auto number_format_id = source_format.d_->number_format_id.get();
+        xlnt::number_format src_number_format;
+
+        if (number_format::is_builtin_format(number_format_id))
+        {
+            src_number_format = number_format::from_builtin_id(number_format_id);
+        }
+        else
+        {
+            const auto it = std::find_if(source_format.d_->parent->number_formats.begin(),
+                source_format.d_->parent->number_formats.end(),
+                [number_format_id](const xlnt::number_format &nf) {
+                    return nf.id() == number_format_id;
+                });
+            if (it != source_format.d_->parent->number_formats.end())
+            {
+                src_number_format = *it;
+            }
+        }
+
+        cloned_format.number_format(src_number_format, source_format.d_->number_format_applied);
+    }
+
+    // Copy protection if present
+    if (source_format.d_->protection_id.is_set())
+    {
+        auto src_protection = source_format.d_->parent->protections.at(source_format.d_->protection_id.get());
+        cloned_format.protection(src_protection, source_format.d_->protection_applied);
+    }
+
+    // Copy other properties
+    if (source_format.d_->pivot_button_)
+    {
+        cloned_format.pivot_button(source_format.d_->pivot_button_);
+    }
+
+    if (source_format.d_->quote_prefix_)
+    {
+        cloned_format.quote_prefix(source_format.d_->quote_prefix_);
+    }
+
+    // Try to copy style name if it exists in destination workbook
+    // This preserves style-based formatting when styles are shared across workbooks
+    if (source_format.d_->style.is_set())
+    {
+        const auto &style_name = source_format.d_->style.get();
+
+        // Only copy if destination workbook has a style with the same name
+        // This avoids creating style name mismatches or conflicts
+        if (workbook().has_style(style_name))
```

Comment:
Better (?) alternative would be perform a deep copy of the style too, if the target workbook doesn't contain the style.
Even if the target workbook contains a style with the given name, one could argue that we should still perform a deep copy of the style if the style with this name doesn't match in the source and target workbook.
Nevertheless, I'm ok with how it is currently implemented for now. I would just document the current behaviour and mention that deep cloning of the style is not yet implemented.